proto/spacemesh/v1/smesher_types.proto

```diff
@@ -4,8 +4,14 @@ option go_package = "github.com/spacemeshos/api/release/go/spacemesh/v1";
 import "spacemesh/v1/types.proto";
 import "google/rpc/status.proto";
 
-message IsSmeshingResponse {
-    bool is_smeshing = 1;
+message SmeshingStatusResponse {
+    enum SmeshingStatus {
+        SMESHING_STATUS_UNSPECIFIED = 0; // Lane's favorite impossible value
+        SMESHING_STATUS_IDLE = 1;
```

Comment:
I would change the statuses to be more descriptive of the init process flow, like this:
POS_STATUS_NOT_STARTED = 1; // pos init was never executed for this node.
POS_STATUS_IN_PROGRESS = 2; // pos init (labels+pow computation) is in progress.
POS_STATUS_DONE = 3; // pos init is done and is pos data is valid
POS_STATUS_ERROR = 4; // last pos init task failed - user should call trigger init again.

Reply:
```
POS_STATUS_NOT_STARTED = 1; // pos init was never executed for this node.
```
We can specify that the status is idle, but not that it was never started, because we have decided to not persist the post init options, and this indication can be determined only against these options.

Reply:
Can we tall if last post setup ended without or with an error? Or not even that?

Reply:
From a user-perspective, the mode detailed the status os - the better. Users want to know what happened to a pos init they started - did it fail? did it complete without an error? Is it in progress? The goal of this api is to provide them as much info about the init process as possible. If more info on pos setup needs to be persisted then I think it will be a great feature to have.

Reply:
Please call it PoST and not PoS. It's confusing enough that there are two different definitions both called "proofs of space" and also "proofs of space and time" in addition to "proofs of spacetime" (that we're using); let's stick with the terminology as published in the literature...

Reply:
If we stay with GetSmeshingStatus() as the name of this method then here are the responses based on my discussion with @moshababo on them:

SMESHING_STATUS_NOT_SETUP = 1;  // node doesn't have a valid pos config to use for setup
SMESHING_STATUS_CREATING_POST_DATA = 2; // pos data init process is in progress
SMESHING_STATUS_POST_DATA_CREATED = 3; // node has a valid pos to use for activations and has finished creating data (init process done)
SMESHING_STATUS_ERROR = 4; // node has pos config info (via config or flags) but pos init has failed.

- The statuses stream should return error or progress (onOf pattern).
- There should be a new method to query the node for pos data init errors. e.g. GetLastError() - it should return status code and message using grpc built-in status in responses.
- It might be good to change GetSmeshingStatus() to GetPosStatus() as the status is only about the proof of space data init  part and not activations or poet related statuses - this data will come from activations meta-data.

Reply:
* “Proof of space” is incorrect, because this phase isn’t about producing a proof. 
* “Proof of space setup” sound like you are referring to a different cryptographic primitive.
* “Proofs of space-time setup” (abbreviated “post-setup”) is a bit more lengthy, but the most accurate and least confusing IMO. Because it’s referring to a “setup”, I don’t see a problem with having the time dimension there. Removing it doesn't make it accurate, and could create more confusion, because users are expected to be familiar with the notion of space-time and the PoST abbreviation, as the protocol’s base cryptographic primitive.

Reply:
I like "post-setup"